packages/lodestar-utils/test/unit/logger/format.test.ts

```diff
@@ -1,21 +1,25 @@
 import {serializeContext} from "../../../src/logger";
 import {expect} from "chai";
-import {config} from "@chainsafe/lodestar-config/lib/presets/minimal";
 
 describe("log format", function () {
 
   describe("serialize context", function () {
     it("should serialize context", function () {
       const serialized = serializeContext({
-        array: config.types.Uint64.hashTreeRoot(10n),
-        buffer: Buffer.alloc(2, 0),
+        object: {
+          foo: "bar",
+          baz: 1,
+        },
```

Comment:
It was designed to handle single level context to force more readable and cleaner logs. If you wish to log object you should serialize it and submit to context. This change makes logger smarter but it could easily result in log clutter.

If you are worried about serializing logs unnecessary for debug logs that aren't displayed we could accept function as context to resolve only if it's going to be displayed

Reply:
I know it was designed that way, I do think we should minimize nested structures when we can. 
I'm just trying to make our logs more machine readable, and handling arbitrary `Json` IMO is part of the process, especially for debug or verbose-level logs.

IMO, in an ideal setting, we would output logs to JSON-LD format in production, or with a certain "productionized logs" flag.
We would still have a pretty-printer, like in this file, for daily use, but the underlying representation is simply `Json`. This will allow us to easily ingest logs into log aggregators and do more advanced analytics on logs. This is how logging will be used in production settings.
Logs that look like `[CHAIN] "Attestation ${X} processed at epoch ${Y}"` should be refactored into `[CHAIN] "processed attestation", {attestationRoot, epoch}`. Which has a machine representation of `{timestamp: 9999999, module: "chain", message: "processed attestation", attestationRoot, epoch}`